providers/flipt/src/main/java/dev/openfeature/contrib/providers/flipt/FliptProvider.java

```diff
@@ -0,0 +1,240 @@
+    private ProviderState state = ProviderState.NOT_READY;
+
+    private AtomicBoolean isInitialized = new AtomicBoolean(false);
+
+    /**
+     * Constructor.
+     * @param fliptProviderConfig FliptProviderConfig
+     */
+    public FliptProvider(FliptProviderConfig fliptProviderConfig) {
+        this.fliptProviderConfig = fliptProviderConfig;
+    }
+
+    /**
+     * Initialize the provider.
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        boolean initialized = isInitialized.getAndSet(true);
+        if (initialized) {
+            throw new GeneralError("already initialized");
+        }
+        super.initialize(evaluationContext);
+        fliptApiClient = fliptProviderConfig.getFliptApiClientBuilder().build();
+
+        state = ProviderState.READY;
+        log.info("finished initializing provider, state: {}", state);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public void emitProviderReady(ProviderEventDetails details) {
+        super.emitProviderReady(details);
+        state = ProviderState.READY;
+    }
+
+    @Override
+    public void emitProviderError(ProviderEventDetails details) {
+        super.emitProviderError(details);
+        state = ProviderState.ERROR;
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+
+        Map<String, String> contextMap = buildContextMap(ctx);
+        EvaluationRequest request = EvaluationRequest.builder().namespaceKey(fliptProviderConfig.getNamespace())
+            .flagKey(key).entityId(ctx.getTargetingKey()).context(contextMap).build();
+
+        BooleanEvaluationResponse response = null;
+        try {
+            response = fliptApiClient.evaluation().boolean_(request);
+        } catch (Exception e) {
+            log.error("Error evaluating boolean", e);
+            return ProviderEvaluation.<Boolean>builder()
+                .value(defaultValue)
+                .errorMessage(e.getMessage())
+                .build();
+        }
+
+        return ProviderEvaluation.<Boolean>builder()
+            .value(response.getEnabled())
+            .reason(response.getReason().toString())
+            .build();
+    }
+
+    private static Map<String, String> buildContextMap(EvaluationContext ctx) {
+        Map<String, String> contextMap = new HashMap<>();
+        ctx.asObjectMap().forEach((k, v) -> {
+            contextMap.put(k, String.valueOf(v));
+        });
+        return contextMap;
+    }
```

Comment:
This method seems to be placed in a bit of an odd spot (mixed in with the evaluation methods).

Reply:
Handled